core/localizer.js

```diff
@@ -497,120 +574,124 @@ var Localizer = exports.Localizer = Montage.specialize( /** @lends Localizer.pro
+                    self.initWithLocale();
+
+                    return self.messagesPromise.then(function () {
+                        return self.localize(key, defaultMessage, defaultOnFail, callback);
+                    });
+                });
+
+            } else if (!this.messagesPromise) {
                 promise = Promise.resolve(this.localizeSync(key, defaultMessage));
                 promise.then(callback);
                 return promise;
-            }
-
-            var l = function () {
-                var messageFn = self.localizeSync(key, defaultMessage);
-                if (typeof callback === "function") {
-                    callback(messageFn);
-                }
-                return messageFn;
-            };
 
-            if (defaultOnFail) {
-                // Try and localize the message, no matter what the outcome
-                return this.messagesPromise.then(l, l);
             } else {
-                return this.messagesPromise.then(l);
+                var l = function () {
+                        var messageFn = self.localizeSync(key, defaultMessage);
+                        if (typeof callback === "function") {
+                            callback(messageFn);
+                        }
+                        return messageFn;
+                    };
+
+                if (defaultOnFail) {
+                    // Try and localize the message, no matter what the outcome
+                    promise = this.messagesPromise.then(l, l);
+
+                } else {
+                    promise = this.messagesPromise.then(l);
+                }
             }
+
+            return promise;
         }
-    }
+    },
 
-});
+    /**
+     * Reset the saved locale back to default by using the steps above.
+     * @function
+     * @returns {string} the reset locale
+     */
+    reset: {
+        value: function () {
+            if (this.shouldStoreLocale && typeof window !== "undefined" && window.localStorage) {
+                window.localStorage.removeItem(LOCALE_STORAGE_KEY);
+            }
 
-/**
- * @class DefaultLocalizer
- * @extends Localizer
- */
-var DefaultLocalizer = Localizer.specialize( /** @lends DefaultLocalizer# */ {
+            this.initWithLocale();
 
-    init: {
-        value: function () {
-            var defaultLocale = this.callDelegateMethod("getDefaultLocale");
+            return this._locale;
+        }
+    },
 
-            if (!defaultLocale && typeof window !== "undefined") {
-                if (window.localStorage) {
-                    defaultLocale = window.localStorage.getItem(LOCALE_STORAGE_KEY);
+
+    _dispatchLocaleChange: {
+        value: function (previousLocale, _component) {
+            if (!_component) {
+                _component = this.component || rootComponent;
+
+                if (typeof _component.LocalizerDidChangeLocale === "function") {
+                    _component.LocalizerDidChangeLocale(this, previousLocale, this._locale);
```

Comment:
this should be localizerDidChangeLocale, lowercase "l". Symbol names starting by a capital letter are usully reserved for Types/classes, constants